### ISSUE TITLE
Show name of road/exit number for next turn while driving

### ISSUE THREAD

**Issue by TriggerDingus** [organicmaps/organicmaps#5111]
**Is your feature request related to a problem? Please describe.**
The issue is while I'm driving, I never know what road or exit to be watching for because Organic Maps doesn't tell me; all it shows is how far away my next turn is. That also becomes a problem because when I don't know what road to be watching for, I sometimes miss my turn because Organic Maps often doesn't tell me to turn/exit until I'm on top of the turn/exit, which is often too late.

Capture of Organic Maps navigation, showing lack of road name/number at next turn.
![image](https://github.com/organicmaps/organicmaps/assets/75340371/f0a2f902-28de-47a5-a768-f9c9357399e0)

**Describe the solution you would like**
Like a Garmin navigator unit, show the name or number of the next turn at the top of the screen so I'm always aware what to watch for while driving.

**Describe alternatives you have considered**
I sometimes forego Organic Maps in favor of my Garmin unit, however, I'd very much like to forego my Garmin unit since Organic Maps is much handier when I know it will work for my purpose.

**Additional context**
Attached, see snapshot of a Garmin navigation screen with name of road to turn on
![cap](https://github.com/organicmaps/organicmaps/assets/75340371/9850233b-12f2-4ede-ad14-db8be9ae348f)
.

**Comment by TriggerDingus** [comment]
It appears the latest version from F-Droid, as indicated by my App Info, is 2023.09.22-27.FDroid.
Is that a suitable version to be testing with?

**Comment by zyphlar** [comment]
@TriggerDingus yes we've had this feature for almost a year I think. A new version should be released before the end of the year that also announces the name of the next turn verbally like "in 1000 feet, turn left onto Main Street".

**Comment by TriggerDingus** [comment]
OK - I'll give it a test drive in the next few days and let you know. I entered the issue in May and you mentioned this feature was added almost a year ago, so I'm curious if I'll see a difference. I keep my software pretty well updated so I don't think I would have been using an old version, previously.

**Comment by zyphlar** [comment]
@TriggerDingus should look like this (in day mode it's black text on a white banner)
![Screenshot_20231112_181356_Organic Maps (Debug)](https://github.com/organicmaps/organicmaps/assets/48434/e89e177e-6b07-4da4-a2ec-347a2d0f20c7)

**Comment by TriggerDingus** [comment]
I tested it today again and it doesn't act any different than when I logged the issue.
It does show the street name, eventually, but as described in my issue it can be too late when you're on a four-lane interstate and the announcement is made that you need to exit in 1000 feet...when you are in the far left lane on a crowded interchange.

I'm curious - is there a technical challenge to keeping my turn/exit name at the top always, right along the distance to that next turn/exit? That is valuable information, especially when I'm traveling long distance on the interstate - I like to look down to confirm my direction of travel (which I created separate request for), distance to my next exit, name of next exit. It could be hours until I reach that exit and so seeing my exit is coming up in 10 miles (or whatever) lets me know it's time to start watching exit signs....and within 1-2 miles to make sure I'm in the proper lane to exit.  

I think the way it currently works may be fine for densely populated places because the environment is very different.

**Comment by zyphlar** [comment]
Gotcha so the name is shown for you, you'd just rather it be shown longer (or maybe always.)

I think that's a reasonable request, and the big question is just what's most reasonable for most people. We wouldn't want the screen to get too cluttered with information that won't be relevant for 100 miles, but obviously we don't want to cause people to be chaotically crossing lanes at the last minute either. I agree that OM could do a better job of telling you stuff like "continue for 235 miles" in addition to stuff like "in one mile, take exit 123."

The distance should be dynamic based on speed, could you tell me what a typical exit announcement for you is, about how fast you were going, and how far you'd have preferred it to be instead? I find "one mile" to be a good distance personally and I know sometimes it's shorter but I'm curious your experience.

**Comment by TriggerDingus** [comment]
I've had turn/exit announcements come as late as 500-1000 feet at anywhere from 65-75mph.
I might suggest adding an option to have the exit/distance information displayed all the times? The only reason I'm using an app such as OrganicMaps or my Garmin unit is to see that information so it helps me to navigate. If it isn't immediately accessible, then it diminishes the usefulness of a navigation app (for my purposes anyway). 

I'm not sure if you care, but the Garmin unit displays this turn information at all times. If the Garmin isn't in navigation mode, that space at the top always displays the next oncoming road name, so you are always aware the intersection ahead of you. This advantage is what keeps me using the Garmin unit.

**Comment by zyphlar** [comment]
Gotcha thanks for the detail! I agree I could use a little more hand-holding from Organic Maps so the question is how much is best for everyone / do we have an option to show it always or what. Maybe others can chime in.

**Comment by biodranik** [comment]
@TriggerDingus Do you hear directions using Bluetooth? Some users reported delays caused by their Bluetooth systems. Or is it only about the time when directions are displayed on the screen?

If your issue is caused by late directions, then the right fix is to hear/see them earlier.

**Comment by TriggerDingus** [comment]
This wouldn't be a bluetooth issue in my case.
I'm not sure was is inferred by "right fix is to hear/see them earlier". This is why I logged the issue, because I want to hear them earlier and see them always.

You can close the issue if you don't agree with my suggestion. It sounds like you prefer it the way it is instead of showing navigation information at all times, which is certainly your prerogative. I'll stick with my navigation unit since that's what it's designed for - perhaps Organic Maps was intended for a different use case than mine and that's fine too.

**Comment by biodranik** [comment]
Why did you close the issue? Let's check what is wrong and improve OM. Am I right that you never see the street name where you are turning to? That's very strange because this feature was implemented more than a year ago. Can you please provide some test routes where it doesn't work?

**Comment by zyphlar** [comment]
@biodranik he does, he just wants the next street name to be always shown / shown much more. I think it's a good idea (GMaps for example will announce "continue for 20 miles" immediately after a turn onto a highway which is reassuring. If you haven't heard from the GPS in awhile it is anxiety inducing, and knowing in advance "when I see Rural Road I'll exit" even if it's dozens or hundreds of miles away can be a good thing. My main thought is, to what degree is appropriate and should it be an option

**Comment by biodranik** [comment]
I'm not aware of any major navigation apps that show the turn lane 100 miles in advance. It is an unnecessary information. You may never drive to that point. 

Announcing "turn left and continue for 100 miles" is fine, it brings a lot of value for a driver.

**Comment by zyphlar** [comment]
@biodranik not turn lane but turn street name. Trigger says that his Garmin displays the next street name at all times even when navigation is off. I agree it's probably more than necessary for a lot of people but it is common, my Toyota in-car Alpine GPS does similar things to contextualize where you are and where you're headed, like "gas/food/hotel available at next highway exit" for the next 4 exits etc. Consider the case which has happened far too often for me: I turn my car radio off but Bluetooth is still connected so my phone doesn't make any noise. I've missed plenty of turns that way, having "exit 123" in my mind for many minutes or even hours helps provide additional safety.

In general Google Maps makes me feel guided and OM makes me feel a little uneasy, the very minimum result of this issue I think should be increasing the maximum turn notification distance (and name display distance) to 1 mile on highways and 1/4 mile on streets. It can be too short to maneuver otherwise if not looking at the map and preparing. I echo his experience that 500-1000ft is almost dangerous and definitely frustrating at 60-70mph.

**Comment by biodranik** [comment]
If maneurs are announced too late, that of course should be fixed by announcing them earlier.

**Comment by biodranik** [comment]
Let's focus on the original issue first and confirm it. @TriggerDingus am I correctly understand that you don't see the turn street name until it's too late (e.g. a few seconds to maneur)? In other words:
- The turn street name is not displayed at the top of the screen if the distance to the maneuver is too long
- The turn street name is displayed at the top of the screen when the maneuver is nearby, but it's too late for you.

Correct?
